### ISSUE TITLE
Semantic release

### ISSUE THREAD

**Issue by maoo** [symphonyoss/python-symphony#35]
In order to publish Python packages on pypi on behalf of the symphonyoss user, I would like to investigate http://python-semantic-release.readthedocs.io/en/latest/automatic-releases/travis.html , as it would give the opportunity to:
1. automate deployment and release for all Python hosted projects
2. publish packages on behalf of symphonyoss user, but avoid to distribute credentials

@openfly - I would like have your opinion on this and if you think it's a good idea; if so, we could:
1. create a `dev` branch, where the "heavy" development takes place
2. enable semantic-release only on master branch, leading to a release on every merge (rather than any commit)

The [hubot-symphony](https://github.com/symphonyoss/hubot-symphony) project already uses this approach (for NodeJS) and works quite well; of course, the same can be easily configured for the [metronome](https://github.com/symphonyoss/metronome) project.

I have already configured step 2, which is the only part where Foundation Staff is required.

**Comment by openfly** [comment]
Not sure that automated deployment is a good idea.  For a variety of reasons.  The first being that unit / functional testing is in no way up to snuff yet.

**Comment by openfly** [comment]
at the end of the day, human eyes are still better than a computers for some things... and i think cutting of releases may be better done by people.

**Comment by maoo** [comment]
Thanks @openfly for the answer; I can safely close this issue, as there's nothing to do; it's totally fine (and understandable) if you prefer to cut releases from your local workstation, the way you always did.

That said, I want to drop few notes based on your comments/questions, as I believe there will be more discussions like this in the future and this content will definitely help; feel free to continue the discussion even if the issue is closed.

Why automated releases are a good idea? In a nutshell, to force best practices; the longer answer can be found [on semantic-release.org docs](https://github.com/semantic-release/semantic-release#is-it-really-a-good-idea-to-release-on-every-push)

The fact that your code is not fully covered by tests is somehow expected, since this is a young project, but it doesn't interfere with automatic releases, since the releases are only cut from one specific (preferably "stable") branch; for example, the [hubot-symphony](https://www.npmjs.com/package/hubot-symphony) project cut a release on every PR on master branch; the same (master) [branch is protected](https://help.github.com/articles/about-protected-branches/) from direct commits, therefore only PRs are accepted.

These 2 simple configurations allow to release anytime you want, by simply submitting PRs from the "working" branch to the "stable" one (may be master or any other branch); this process is still done by humans, who can do all the checks they want before proceeding.

To summarise, the main difference between automated and manual release is that the former is executed from a remote server and the latter is executed from a local workstation; all the other aspects can be configured.

The Foundation is in the process of documenting the different release flavours that are available for each language/eco-system, so I thought that pyhton-symphony could have been a good one to tackle this challenge with Python.

Thank you very much for your feedback!

**Comment by openfly** [comment]
Yeah it's an interesting discussion idea.  From the security perspective you'd generally want a release manager or set of with specific signing keys... then you'd want an index signing key from your automated release.  

Sadly pypi is not really package management and almost as bad as npm as far as security goes.  So people tend not to follow normal conventions.

Packaging and automation of the building thereof will need some work in the near and distant future.  Just not there yet.
